### PR TITLE
feat(firebase): delete user and throw errors correctly

### PR DESCRIPTION
**Pull request by christianalfoni**
- Can not return values from `catch` as they will be no longer propagate as errors, have to reject or throw again

- Wrapped a few methods with normal promises, cause Firebase promise is not native

### CODE REVIEW

File: packages/cerebral-provider-firebase/src/deleteUser.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import firebase from 'firebase'
+
+export default function deleteUser (password) {
+  return new Promise((resolve, reject) => {
+    const user = firebase.auth().currentUser
+
+    return user.delete().then(() => resolve(), (error) => reject({error}))
```
Comment (edgesoft) [inline packages/cerebral-provider-firebase/src/deleteUser.js]: Can user be null here? According to docs "You can also get the currently signed-in user by using the currentUser property. If a user isn't signed in, currentUser is null:"

File: packages/cerebral-provider-firebase/src/sendPasswordResetEmail.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import firebase from 'firebase'
+
+export default function sendPasswordResetEmail (email) {
+  return new Promise((resolve, reject) => {
+    firebase.auth().sendPasswordResetEmail(email)
+      .then(() => resolve())
+      .catch((error) => reject({error}))
```
Comment (gaspard) [inline packages/cerebral-provider-firebase/src/sendPasswordResetEmail.js]: This should be (catch does not work with firebase own Promises):

```js
 firebase.auth().sendPasswordResetEmail(email)
      .then(
        () => resolve(),
        (error) => reject({error})
      )
```

Comment (christianalfoni) [inline packages/cerebral-provider-firebase/src/sendPasswordResetEmail.js]: Great catch, thanks! :)

**Comment by christianalfoni** [comment]
Okay, this should now be fixed... We just realized Google has updated Firebase and there is a new approach to authentication on the backend side of things. We need to pass the token on every Task to identify the user on the server. A small detail, but necessary :)

File: packages/cerebral-provider-firebase/src/createTask.js [hunk 3]
```diff
@@ -5,20 +5,25 @@ export default function createTask (options) {
...
+      return firebase.auth().currentUser.getToken()
+        .then((_token) => {
+          const taskKey = ref.push(Object.assign({
+            _state: options.specPrefix ? `${options.specPrefix}_${name}` : name,
+            _token
+          }, payload))
+
+          const taskRef = firebase.database().ref(`${tasksPath}/${taskKey.key}`)
+          taskRef.on('value', data => {
+            if (!data.val()) {
```
Comment (gaspard) [inline packages/cerebral-provider-firebase/src/createTask.js]: maybe add:

```js
const value = data.val()
```

Not sure about the cost of "data.val()"... But maybe it's very small.

**Pull request merged**